server/authentication/auth.js

```diff
@@ -1,16 +1,18 @@
 const passport = require('passport')
 const { Strategy } = require('passport-oauth2')
+const querystring = require('querystring')
 const config = require('../config')
 const { generateOauthClientToken } = require('./clientCredentials')
 
-function authenticationMiddleware() {
-  return (req, res, next) => {
-    if (req.isAuthenticated()) {
+function authenticationMiddlewareFactory(tokenVerifier) {
+  return async (req, res, next) => {
+    if (req.isAuthenticated() && (await tokenVerifier.verify(res.locals.user.token))) {
       return next()
     }
-
-    req.session.returnTo = req.originalUrl
-    return res.redirect('/login')
+    req.logout()
+    const query = querystring.stringify({ returnTo: req.originalUrl })
+    res.redirect(`/login?${query}`)
+    return undefined // because typescript insists on a return value
```

Comment:
kind of standard to just `return res.redirect(`/login?${query}`)` in express apps

Reply:
Done